EIPS/eip-2028.md

```diff
@@ -0,0 +1,77 @@
+---
+eip: 2028
+title: Calldata gas cost reduction
+author: Alexey Akhunov(@AlexeyAkhunov), Eli Ben Sasson (eli@starkware.co), Tom Brand (tom@starkware.co), Avihu Levy (avihu@starkware.co)
+discussions-to: https://ethereum-magicians.org/t/eip-2028-calldata-gas-cost-reduction/3280
+status: Draft
+type: Standards Track
+category: Core
+created: 2019-05-03
+---
+
+## Simple Summary
+We propose to reduce the gas cost of Calldata (`GTXDATANONZERO`) from its current value of 68 gas per byte to a lower cost, to be backed by mathematical modeling and empirical estimates. The mathematical model is the one used in the works of Sompolinsky and Zohar [1] and Pass, Seeman and Shelat [2], which relates network security to network delay. We shall (1) evaluate the theoretical impact of lower Calldata gas cost on network delay using this model, (2) validate the model empirically, and (3) base the proposed gas cost on our findings.
+
+## Motivation
+There are a couple of main benefits to accepting this proposal and lowering gas cost of Calldata
+On-Chain Scalability: Generally speaking, higher bandwidth of Calldata improves scalability, as more data can fit within a single block.
+Layer two scalability: Layer two scaling solutions can improve scalability by moving storage and computation off-chain, but often introduce data transmission instead.
+Proof systems such as STARKs and SNARKs use a single proof that attests to the computational integrity of a large computation, say, one that processes a large batch of transactions.
+Some solutions use fraud proofs which requires a transmission of merkle proofs.
+Moreover, one optional data availability solution to layer two is to place data on the main chain, via Calldata.
+Stateless clients: The same model will be used to determine the price of the state access for the stateless client regime, which will be proposed in the State Rent (from version 4). There, it is expected that the gas cost of state accessing operation will increase roughly proportional to the extra bandwidth required to transmit the “block proofs” as well as extra processing required to verify those block proofs.
+
+## Specification
+`GTXDATANONZERO` is currently 68 gas per byte.
```

Comment:
I think this should be more explicit, since we do not have a canonical spec and G<sub>txdatanonzero</sub> is just a notation used in the Yellow Paper.

I'd suggest to explain that "The gas per non-zero byte is reduced from 68 to n. Gas cost of zero bytes is unchanged.`

Reply:
Thanks, good suggestion.